lib/biker.rb

```diff
@@ -1,3 +1,27 @@
 class Biker
+  attr_reader :name,
+              :rides,
+              :max_distance,
+              :acceptable_terrain
+
+  def initialize(name, max_distance)
+    @name = name
+    @rides = Hash.new([])
```

Comment:
I sent you astray with this! Thought it would work better but I got some weird behavior doing this. Instead, you could've left it as a a `{}` and then added an if statement in your log_ride in order to check if the key existed, and if it didn't, set the key to an empty array to avoid those nil errors.

Reply:
Another cool option is the `fetch` method on hash, which allows you to pass a default value if the key doesn't exist.